app/src/lib/git/rebase.ts

```diff
@@ -82,6 +86,73 @@ export async function getRebaseContext(
 
   return null
 }
+
+/** Regex for identifying when rebase applied each commit onto the base branch */
+const rebaseApplyingRe = /^Applying: (.*)/
+
+/**
+ * A parser to read and emit rebase progress from Git `stdout`
+ */
+class GitRebaseParser {
+  private currentCommitCount = 0
+
+  public constructor(
+    startCount: number,
+    private readonly totalCommitCount: number
+  ) {
+    this.currentCommitCount = startCount
+  }
+
+  public parse(line: string): IRebaseProgress | null {
+    const match = rebaseApplyingRe.exec(line)
+    if (match == null || match.length !== 2) {
```

Comment:
Can we make the null check a triple equals please?

```suggestion
    if (match === null || match.length !== 2) {
```